QualificationsRegister/Services/Database/RegisterDb.cs

```diff
@@ -1,37 +1,106 @@
+using Azure;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
 using Ofqual.Common.RegisterAPI.Models.DB;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using static System.Runtime.InteropServices.JavaScript.JSType;
 
 namespace Ofqual.Common.RegisterAPI.Services.Database
 {
-    public class RegisterDb : IRegisterDb
+    public partial class RegisterDb : IRegisterDb
```

Comment:
why is it partial?

Reply:
That's because the regex object needs to be declared as partial - https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators

Reply:
More details here - https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/